### DISCUSSION TITLE
Support for the Shelly Pro 3em?

### DISCUSSION THREAD

**Discussion by NickRoberts174** [fabian-lauer/dbus-shelly-3em-smartmeter#28] [Ideas]
Hi there, Recently upgraded from my shelly em (single phase) to the shelly pro 3em (to monitor 3 circuits on the same phase - not to monitor 3 phases). I had hoped this integration would work off the bat, unfortunately not :( 

Is there any tweak that can be made to get this up and running? The Pro 3em has a ton more funtionality than the standard 3em including modbus, LAN support, supports mjS scripting etc so it would b cool to see this added!

**Comment by Christian-Fecher** [comment]
Hello everyone,

I have the Shelly Pro 3EM running with this base here from Fabian.

I made some changes to the python script.
This means that the Shelly Pro 3EM has been running over the LAN interface for a week now.

That was also the reason why I wanted to use the Shelly Pro 3EM. The LAN interface.

I'll post my code adjustments as an issue here in Fabian's repo.

The adjustments are on the
URL for the GET request
and
The JSON file is completely different than the Shelly 3EM, so reading the JSON file has to be adjusted.

**Comment by fabian-lauer** [comment]
Hi,

as an short update - there is a discussion/issue ongoing here: https://github.com/fabian-lauer/dbus-shelly-3em-smartmeter/discussions/42

Its in German, but the guys have solved the problem and also and updated script available.

Cheers
